<?xml version="1.0"?>
<robot name="ur10e_robot" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!-- Include the UR10 robot description -->
  <xacro:include filename="$(find nrs_path)/urdf/ur10e.xacro" />
  <!-- Define the testbed link -->
  <link name="testbed">
    <visual>
      <origin xyz="0.3 0.5 -0.45" rpy="0 0 0" />
      <geometry>
        <box size="0.84 1.24 0.9" /> <!-- 840mm x 1240mm x 900mm -->
      </geometry>
      <material name="testbed_gray">
        <color rgba="0.5 0.5 0.5 0.5" />
      </material>
    </visual>
    <collision>
      <origin xyz="0.3 0.5 -0.45" rpy="0 0 0" />
      <geometry>
        <box size="0.84 1.24 0.9" />
      </geometry>
      <material name="testbed_gray">
        <color rgba="0.5 0.5 0.5 0.5" />
      </material>
    </collision>
    <inertial>
      <mass value="100" />
      <origin xyz="0.3 0.5 -0.45" rpy="0 0 0" />
      <inertia ixx="33.33" ixy="0.0" ixz="0.0" iyy="33.33" iyz="0.0" izz="33.33" />
    </inertial>
  </link>


  <link name="UR10e_roboworld_tooltip_m">
    <visual>
      <geometry>
        <mesh filename="package://nrs_path/mesh/UR10e_roboworld_tooltip_m.stl" />
      </geometry>
      <material name="tooltip_gray">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://nrs_path/mesh/UR10e_roboworld_tooltip_m.stl" />
      </geometry>
      <material name="tooltip_gray">
        <color rgba="0.5 0.5 0.5 1.0" />
      </material>
    </collision>
  </link> 


  <link name="workpiece">
    <visual>
      <geometry>
        <mesh filename="package://nrs_path/mesh/workpiece.stl" />
      </geometry>
      <material name="light_blue">
        <color rgba="0.57 0.81 0.92 0.5" />
      </material>
    </visual>
    <!-- <collision>
      <geometry>
        <mesh filename="package://nrs_path/mesh/fenda_wrap.stl" />
      </geometry>
      <material name="light_blue">
        <color rgba="0.57 0.81 0.92 0.5" />
      </material>
    </collision> -->
  </link>

  <!-- Define a fixed joint to attach the testbed to the base link -->
  <joint name="testbed_joint" type="fixed">
    <parent link="base_link" />
    <child link="testbed" />
    <origin xyz="0 0 0" rpy="0 0 -0.7853981633974483" /> <!-- 45 degrees in radians -->
  </joint>

  <!-- Define a fixed joint to attach the tooltip to the wrist_3_link -->
  <joint name="tooltip_joint" type="fixed">
    <parent link="wrist_3_link" />
    <child link="UR10e_roboworld_tooltip_m" />
    <origin xyz="0 0 0.0" rpy="0 0 0" />
  </joint>


  <!-- Define a fixed joint to attach the workpiece to the base link -->
  <joint name="workpiece_joint" type="fixed">
    <parent link="base_link" />
    <child link="workpiece" />
    <origin xyz="0.0 0.0 0.0" rpy="0 0 0" />
  </joint>

</robot>